superset/common/query_object.py

```diff
@@ -166,11 +166,11 @@ def __init__(
         #   1. 'metric_name'   - name of predefined metric
         #   2. { label: 'label_name' }  - legacy format for a predefined metric
         #   3. { expressionType: 'SIMPLE' | 'SQL', ... } - adhoc metric
-        self.metrics = [
-            metric
-            if isinstance(metric, str) or "expressionType" in metric
-            else metric["label"]  # type: ignore
-            for metric in metrics
+        self.metrics = metrics and [
+            x
```

Comment:
Use `x` in case some may wonder whether `metric` is also from the parent context (e.g. an `__init__` argument).

Reply:
nit: I'd personally prefer `metric_` here

Reply:
I always prefer `x` in list comprehensions and lambda functions because it's simpler and reads more "local".